packages/redux/src/raffles/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,367 @@
+import * as actionTypes from '../actionTypes';
+import {
+  mockCreateRaffleParticipationsNormalizedPayload,
+  mockFetchRafflesNormalizedPayload,
+  mockRaffleEstimationResponse,
+  mockRaffleParticipationResponse,
+  mockRaffleResponse,
+  participationId,
+  raffleId,
+} from 'tests/__fixtures__/raffles';
+import { toBlackoutError } from '@farfetch/blackout-client';
+import cloneDeep from 'lodash/cloneDeep';
+import reducer, {
+  entitiesMapper,
+  getAllRaffles,
+  getEstimations,
+  getParticipationCreations,
+  getParticipations,
+  getRaffles,
+  INITIAL_STATE,
+} from '../reducer';
+import type { RafflesState } from '../types';
+
+let initialState: RafflesState;
+const randomAction = { type: 'this_is_a_random_action' };
+
+describe('raffles reducer', () => {
+  beforeEach(() => {
+    initialState = INITIAL_STATE;
+  });
+
+  describe('when its the first time it is invoked and receive an undefined state', () => {
+    it('should return the initial state', () => {
+      const state = reducer(undefined, randomAction);
+      expect(state).toEqual(initialState);
+    });
+  });
+
+  describe('resetRafflesState', () => {
+    it('should return the initial state', () => {
+      const state = reducer(initialState, {
+        type: actionTypes.RESET_RAFFLES_STATE,
+      });
+      expect(state).toEqual(initialState);
+    });
+  });
+
+  describe('allRaffles reducer', () => {
+    const expectedError = new Error('fetch all raffles error');
+    it(`should handle ${actionTypes.FETCH_RAFFLES_REQUEST} action type`, () => {
+      const mockState = cloneDeep(initialState);
+      const hash = 'hashRaffleQuery';
+      mockState.allRaffles[hash] = {
+        isLoading: true,
```

Comment:
You are not testing with the most representative states for each action. Although this is a valid test, we should strive to at least test with the most representative states. In this case, `isLoading` should be false since when a request is made, most of the time the `isLoading` in store will be false. Having `isLoading` true in state while the result will also set it to true might not uncover bugs with the code (for example if the reducer returned the received state parameter). Please, review all tests for actions and make sure they are using a correct state:

- Testing XXX_REQUEST actions: `isLoading: false, error: null or defined with an error`
- Testing XXX_SUCCESS actions: `isLoading: true, error: null`
- Testing XXX_FAILURE actions: `isLoading: true, error: null`.

This is just to make sure we catch the most common errors while being practical, since the test case you have here should also be tested.

Also, please, make sure you have spacing between the lines as it makes it difficult to read for an old man like me :)

Reply:
Sure, pls double check

Reply:
Thank you, that is exactly how I wanted!